codex-rs/core/tests/suite/approvals.rs

```diff
@@ -1557,3 +1557,138 @@ async fn run_scenario(scenario: &ScenarioSpec) -> Result<()> {
 
     Ok(())
 }
+
+#[tokio::test(flavor = "current_thread")]
+async fn approving_allow_prefix_persists_policy_and_skips_future_prompts() -> Result<()> {
+    let server = start_mock_server().await;
+    let approval_policy = AskForApproval::UnlessTrusted;
+    let sandbox_policy = SandboxPolicy::DangerFullAccess;
```

Comment:
Let's use `ReadOnly` or something that would block the command if it weren't approved? Might have to add `#[cfg(unix)]` to the test (not sure...).